### PR TITLE
DEVTOOLING-79 Add task management workitems resource

### PR DESCRIPTION
**Pull request by PrinceMerluza**
Add task management workitems resource and data source

### CODE REVIEW

File: genesyscloud/task_management_workitem/examples/data-sources/genesyscloud_task_management_workitem/data-source.tf
Comment (dginty4) [inline genesyscloud/task_management_workitem/examples/data-sources/genesyscloud_task_management_workitem/data-source.tf:1]: Is this file needed?

File: genesyscloud/task_management_workitem/examples/resources/genesyscloud_task_management_workitem/resource.tf
Comment (dginty4) [inline genesyscloud/task_management_workitem/examples/resources/genesyscloud_task_management_workitem/resource.tf:1]: Same here

File: genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	proxy := newTaskManagementWorkitemProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	workitems, err := proxy.getAllTaskManagementWorkitem(ctx)
+	if err != nil {
+		return nil, diag.Errorf("Failed to get task management workitem: %v", err)
+	}
+
+	for _, workitem := range *workitems {
+		resources[*workitem.Id] = &resourceExporter.ResourceMeta{Name: *workitem.Id}
```
Comment (dginty4) [inline genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go:38]: I think it should be `Name: *workItem.Name` otherwise when exported the name of the resource will be a guid

File: genesyscloud/task_management_workitem/examples/data-sources/genesyscloud_task_management_workitem/data-source.tf
Comment (PrinceMerluza) [reply genesyscloud/task_management_workitem/examples/data-sources/genesyscloud_task_management_workitem/data-source.tf:1]: Oops that's the leftover when I pasted the results from the generator. Those doc files are now in the proper place and generated. Deleted.

File: genesyscloud/task_management_workitem/examples/resources/genesyscloud_task_management_workitem/resource.tf
Comment (PrinceMerluza) [reply genesyscloud/task_management_workitem/examples/resources/genesyscloud_task_management_workitem/resource.tf:1]: Same here as well

File: genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go (hunk 1 above)
Comment (PrinceMerluza) [reply genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go:38]: Fixed

**Review by dginty4 — APPROVED** [review]

File: genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+/*
+The resource_genesyscloud_task_management_workitem.go contains all of the methods that perform the core logic for a resource.
+*/
+
+// getAllAuthTaskManagementWorkitem retrieves all of the task management workitem via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthTaskManagementWorkitems(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := newTaskManagementWorkitemProxy(clientConfig)
```
Comment (charliecon) [inline genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go:29]: Not super important, but I think this should be `getTaskManagementWorkitemProxy`

**Review by charliecon — APPROVED** [review]

File: genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go (hunk 2 above)
Comment (carnellj-genesys) [reply genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go:29]: We should fix this to be consistent.

**Review by carnellj-genesys — APPROVED** [review]
Looks good.  One small fix and then go ahead and merge.

File: genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go (hunk 2 above)
Comment (PrinceMerluza) [reply genesyscloud/task_management_workitem/resource_genesyscloud_task_management_workitem.go]: Fixed
